airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -69,23 +69,25 @@ def _parse_next_page_number(response: requests.Response) -> Optional[int]:
             return dict(parse_qsl(urlparse(next_page).query)).get("page")
         return None
 
-    def backoff_time(self, response: requests.Response) -> int:
+    def backoff_time(self, response: requests.Response) -> Union[int, float]:
         """
         The rate limit is 700 requests per minute
         # monitoring-your-request-activity
         See https://developer.zendesk.com/api-reference/ticketing/account-configuration/usage_limits/
         The response has a Retry-After header that tells you for how many seconds to wait before retrying.
         """
-        retry_after = response.headers.get("Retry-After")
-        if retry_after:
-            return int(retry_after)
-        # the header X-Rate-Limit returns a amount of requests per minute
-        # we try to wait twice as long
-        rate_limit = float(response.headers.get("X-Rate-Limit") or 0)
-        if rate_limit:
-            return (60.0 / rate_limit) * 2
-        # default value if there is not any headers
-        return 60
+        # default value if there is not any header
+        sleep_timeout = 60
+        retry_after = int(response.headers.get("Retry-After") or 0)
+        if retry_after and retry_after > 0:
+            sleep_timeout = int(retry_after)
+        else:
+            # the header X-Rate-Limit returns a amount of requests per minute
+            # we try to wait twice as long
+            rate_limit = float(response.headers.get("X-Rate-Limit") or 0)
+            if rate_limit and rate_limit > 0:
```

Comment:
can it be lower that 0.  f not then just simple check is enough 'If retry_after then'

Reply:
add a same comment before